receiver/awscontainerinsightreceiver/internal/ecsInfo/ecs_task_info.go

```diff
@@ -0,0 +1,134 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ecsinfo
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	httpClient "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight/httpclient"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/host"
+)
+
+type ecsTaskInfoProvider interface {
+	getRunningTaskCount() int64
+	getRunningTasksInfo() []ECSTask
+}
+
+type ECSContainer struct {
+	DockerID string
+}
+type ECSTask struct {
+	KnownStatus string
+	ARN         string
+	Containers  []ECSContainer
+}
+
+type ECSTasksInfo struct {
+	Tasks []ECSTask
+}
+
+type taskInfo struct {
+	logger                  *zap.Logger
+	httpClient              httpClient.Requester
+	refreshInterval         time.Duration
+	ecsTaskEndpointProvider hostIPProvider
+	runningTaskCount        int64
+	runningTasksInfo        []ECSTask
+	readyC                  chan bool
+	sync.RWMutex
+}
+
+type taskInfoOption func(*taskInfo)
+
+func newECSTaskInfo(ctx context.Context, ecsTaskEndpointProvider hostIPProvider,
```

Comment:
All of the same comments from the instance info file seem to apply here as well.